test/Basics/VerifyParserState.js

```diff
@@ -0,0 +1,195 @@
+//-------------------------------------------------------------------------------------------------------
+// Copyright (C) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+//-------------------------------------------------------------------------------------------------------
+
+// Following lines generate a baseline with flags:
+// -ForceCreateParserState -trace:CreateParserState
+//
+// Uses WScript.LoadScript so that function numbering can be easier to reason about
+
+console.log('nested function capturing var decl');

```

Comment:
do we need to test with function with eval and with?

Reply:
and with function name expression? 

---
In reply to: [181448145](https://github.com/Microsoft/ChakraCore/pull/4983#discussion_r181448145) [](ancestors = 181448145)

Reply:
Param scope would be good too since it takes different paths with pidrefstack

Reply:
Added unit tests for these cases, thanks for the feedback